src/test/librbd/test_Utils.cc

```diff
@@ -0,0 +1,52 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "test/librbd/test_fixture.h"
+#include "test/librbd/test_support.h"
+#include "librbd/ImageCtx.h"
+#include "librbd/Utils.h"
+
+using librbd::util::data_object_name;
+
+void register_test_utils() {
+}
+
+struct TestUtils : public TestFixture {
+    librbd::ImageCtx *ictx;
+
+    void SetUp() override {
+      TestFixture::SetUp();
+      ASSERT_EQ(0, open_image(m_image_name, &ictx));
+    }
+
+    void TearDown() override {
+      close_image(ictx);
+      TestFixture::TearDown();
+    }
+
+    void expect_object_name(uint64_t object_no,
+                            const std::string& expected_object_name_postfix) {
+      EXPECT_STREQ(
+              (ictx->object_prefix + std::string(".") +
+               expected_object_name_postfix).c_str(),
+              data_object_name(ictx, object_no).c_str());
+    }
+};
+
+TEST_F(TestUtils, test_data_object_name) {
+  ictx->old_format = false;
```

Comment:
Instead of messing with `old_format`, let's make two tests: `DataObjectNameV1` and `DataObjectNameV2` with the corresponding `REQUIRE_FORMAT_V1()` and `REQUIRE_FORMAT_V2()` annotations:
```suggestion
  REQUIRE_FORMAT_V2();

```